sktime/forecasting/online_ensemble.py

```diff
@@ -0,0 +1,150 @@
+import numpy as np
+import pandas as pd
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sktime.forecasting.compose._ensemble import EnsembleForecaster
+from sktime.forecasting.model_selection import SlidingWindowSplitter
+from sktime.utils.validation.forecasting import check_cv
+from sktime.utils.validation.forecasting import check_y
+
+
+class OnlineEnsembleForecaster(EnsembleForecaster):
+    """Online Updating Ensemble of forecasters
+
+    Parameters
+    ----------
+    ensemble_algorithm : ensemble algorithm
+    forecasters : list of (str, estimator) tuples
+    n_jobs : int or None, optional (default=None)
+        The number of jobs to run in parallel for fit. None means 1 unless
+        in a joblib.parallel_backend context.
+        -1 means using all processors.
+    """
+
+    _required_parameters = ["forecasters"]
+
+    def __init__(self, forecasters,
+                 ensemble_algorithm=None,
+                 n_jobs=None):
+
+        self.n_jobs = n_jobs
+        self.ensemble_algorithm = ensemble_algorithm
+        self.weights = np.ones(len(forecasters))/len(forecasters)
+
+#         if self.ensemble_algorithm.n != len(forecasters):
+#             raise ValueError("Number of Experts in Ensemble Algorithm \
+#                              doesn't equal number of Forecasters")
+
+        super(EnsembleForecaster, self).__init__(forecasters=forecasters,
+                                                 n_jobs=n_jobs)
+
+    def fit(self, y_train, fh=None, X_train=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y_train : pd.Series
+            Target time series to which to fit the forecaster.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        X_train : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+
+        self._set_y_X(y_train, X_train)
+        self._set_fh(fh)
+        names, forecasters = self._check_forecasters()
+        self._fit_forecasters(forecasters, y_train, fh=fh, X_train=X_train)
```

Comment:
I think we need to use `self.fh` to get the validated/converted fh
```suggestion
        self._fit_forecasters(forecasters, y_train, fh=self.fh, X_train=X_train)
```

Reply:
actually each member of the ensemble will do the same checks, but they'll be a bit quicker if it has been converted to the right format already

Reply:
That makes sense, will convert to self.fh.

Reply:
I found that self.fh might not be set with the `OptionalForecastingHorizonMixin`, I was using the same `fit` from `EnsembleForecaster`. Is it alright if we keep using `fh`.

Reply:
If you need the fh already during fitting, you can you the `RequiredForecastingHorizonMixin` class

Reply:
I don't think we need `fh` during fitting? I'm fairly sure it is being passed as a `None` right now. Which kinds of forecasters might need `fh` during fitting?

Reply:
A few forecasting approaches need fh during fitting, for example the "direct" strategy in reduction to regression, when you train a separate model for each step ahead in the forecasting horizon.

Reply:
I guess would making the class a `RequiredForecastingHorizonMixin` mean that for every time we try to train, we need to specify a `fh` which is not `None`?